src/Cameras/arcRotateCamera.ts

```diff
@@ -891,14 +891,20 @@ export class ArcRotateCamera extends TargetCamera {
                 this._transformedDirection = Vector3.Zero();

```

Comment:
I think `_localDirection` is not used anymore, you should be able to remove it.

Else LGTM (trusting you on the math, I have not really understood them!).